src/main/java/appeng/parts/p2p/ItemP2PTunnelPart.java

```diff
@@ -190,17 +166,29 @@ public ItemStack insertItem(int slot, @Nonnull ItemStack stack, boolean simulate
         @Override
         @Nonnull
         public ItemStack extractItem(int slot, int amount, boolean simulate) {
-            return ItemP2PTunnelPart.this.getInput().getAttachedItemHandler().extractItem(slot, amount, simulate);
+            try (CapabilityGuard input = getInputCapability()) {
+                ItemStack result = input.get().extractItem(slot, amount, simulate);
+
+                if (!simulate) {
+                    queueTunnelDrain(PowerUnits.RF, result.getCount());
+                }
+
+                return result;
+            }
         }
 
         @Override
         public int getSlotLimit(int slot) {
-            return ItemP2PTunnelPart.this.getInput().getAttachedItemHandler().getSlotLimit(slot);
+            try (CapabilityGuard input = getInputCapability()) {
+                return input.get().getSlotLimit(slot);
```

Comment:
Are there any cases (in other mods), in which this could be called during an extract/insert? E.g. to split the request across multiple linked inventories while taking potential limits into account while using it to keep a ratio based on the slot limits?
If so how would this behave?

Reply:
I'm not sure, worst case we can bump the depth limit to 2 or 3. I suggest we do later it if that proves necessary in the future?